### PR TITLE
add channel encryption for ILS, Video and voice

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
Review these:
- http://localhost:3000/voice-calling/develop/media-stream-encryption?platform=react-native
- http://localhost:3000/video-calling/develop/media-stream-encryption?platform=react-native
- http://localhost:3000/interactive-live-streaming/develop/media-stream-encryption?platform=react-native

I have validated the code on android. Please note that the encryption modes `Aes128Gcm2` and `Aes256Gcm2` are returning -2 which is why I have used another encryption mode so that this sample code works as is.

### CODE REVIEW

File: shared/video-sdk/encrypt-media-streams/project-implementation/react-native.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    ```typescript
+    EncryptionMode,
+    ```
+
+3.  **Call the channel encryption method to enable channel encryption**
+
+    To enable channel encryption in your <Vpl k="CLIENT" />, you need to:
+
+    1. Create an object of `EncryptionConfig` class and specify a configuration for the channel encryption. In the configuration, you specify `encryptionMode`, `encryptionKey`, and `encryptionKdfSalt`. `encryptionKdfSalt` is required for `Aes128Gcm2` and `Aes256Gcm2` modes but optional otherwise.
```
Comment (billy-the-fish) [inline shared/video-sdk/encrypt-media-streams/project-implementation/react-native.mdx:25]: Create an `EncryptionConfig` instance. 

modes, it is optional for other encryption modes.

File: shared/video-sdk/encrypt-media-streams/project-implementation/react-native.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    2. Call `enableEncryption` and pass `true` and the `EncryptionConfig` object as parameters to enable media encryption.
```
Comment (billy-the-fish) [inline shared/video-sdk/encrypt-media-streams/project-implementation/react-native.mdx:27]: \<what you are doing> \<how to do it>

To enable media encryption, call `enableEncryption` with your `EncryptionConfig` instance.

File: shared/video-sdk/encrypt-media-streams/project-implementation/react-native.mdx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    ```typescript
+    let encryptionConfig = {
+        encryptionBase64: encryptionBase64,
+        encryptionKey: encryptionKey,
+        encryptionMode: EncryptionMode.Aes128Ecb,
+      };
+    engine.enableEncryption(true, encryptionConfig);
+    ```
+    <Vg k="COMPANY"/> recommends using `Aes128Gcm2` or `Aes256Gcm2` encrypted mode. These two modes support the use of salt for higher security.
```
Comment (billy-the-fish) [inline shared/video-sdk/encrypt-media-streams/project-implementation/react-native.mdx:39]: Please don't:
* Say Agora recommends, always "best practice"
* Put the number of items you are about to talk about. Chances are you will add or remove an item and forget about the number. Keep it simple. 

Best practice is to use  `Aes128Gcm2` or `Aes256Gcm2`. These modes use salt for higher security.

File: shared/video-sdk/encrypt-media-streams/project-test/react-native.mdx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+- *Android*
+
+    1. Enable the Developer options on your Android device, and then connect it to your computer using a USB cable.
```
Comment (billy-the-fish) [inline shared/video-sdk/encrypt-media-streams/project-test/react-native.mdx:3]: , then connect it

File: shared/video-sdk/encrypt-media-streams/project-test/react-native.mdx [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    1. Run npx react-native run-android in the project root directory.
+
+- *iOS:*
+
+    1. Open the ProjectName/ios/ProjectName.xcworkspace folder with Xcode.
+    1. Connect your iOS device to your Mac using a USB cable.
+    1. Click the Build and run button in Xcode.
```
Comment (billy-the-fish) [inline shared/video-sdk/encrypt-media-streams/project-test/react-native.mdx:10]: In Xcode, run your project. 


They are developers, they should have figured this one out :-).

**Review by billy-the-fish — CHANGES_REQUESTED** [review]
Nearly there, good work.

**Comment by Kishan-Dhakan** [comment]
@billy-the-fish, I resolved the comments.

**Review by billy-the-fish — APPROVED** [review]
Good work, thank you.

**Pull request merged**